learning_journal/learning_journal/tests.py

```diff
@@ -0,0 +1,161 @@
+import pytest
+import os
+import transaction
+import datetime
+
+from pyramid import testing
+
+from .models import (
+    MyEntry,
+    get_engine,
+    get_session_factory,
+    get_tm_session
+)
+from .models.meta import Base
+
+DB_SETTINGS = {'sqlalchemy.url': 'postgres://VMB@localhost:5432/mockdb_ljtesting'}
+
+
+@pytest.fixture(scope="function")
+def setup_test_env():
+    os.environ["DATABASE_URL"] = 'postgres://VMB@localhost:5432/mockdb_ljtesting'
+
+@pytest.fixture(scope="function")
+def sqlengine(request):
+    """Set up sql engine."""
+    config = testing.setUp(settings=DB_SETTINGS)
+    config.include(".models")
+    settings = config.get_settings()
+    engine = get_engine(settings)
+    Base.metadata.create_all(engine)
+
+    def teardown():
+        testing.tearDown()
+        transaction.abort()
+        Base.metadata.drop_all(engine)
+
+    request.addfinalizer(teardown)
+    return engine
+
+
+@pytest.fixture(scope="function")
+def new_session(sqlengine, request):
+    """Set up new new session."""
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    def teardown():
+        transaction.abort()
+
+    request.addfinalizer(teardown)
+    return session
+
+
+@pytest.fixture(scope="function")
+def populated_db(request, sqlengine):
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    with transaction.manager:
+        session.add(MyEntry(title="Test Title", body="This is a test entry, James is being awesome.", creation_date=datetime.datetime.utcnow()))
+
+    def teardown():
+        with transaction.manager:
+            session.query(MyEntry).delete()
+
+    request.addfinalizer(teardown)
+
+
+@pytest.fixture()
+def dummy_request(new_session):
+    """Call a dummy request."""
+    return testing.DummyRequest(dbsession=new_session)
+
+
+def test_home_view(dummy_request, new_session):
+    """Test entries are in home view."""
+    from .views.default import home_view
+    new_session.add(MyEntry(title="test", body="this is a test", creation_date=datetime.datetime.utcnow()))
+    new_session.flush()
+    info = home_view(dummy_request)
+    assert "entries" in info
```

Comment:
how many entries are there?  Are they the right ones?  Do you really see all of them?